src/list.cpp

```diff
@@ -254,39 +401,66 @@ util::Optional<Mixed> List::min(size_t column)
     return as_results().min(column);
 }
 
-util::Optional<Mixed> List::sum(size_t column)
+Mixed List::sum(size_t column)
 {
-    return as_results().sum(column);
+    // Results::sum() returns none only for Empty results
```

Comment:
To make sure I understand correctly: you're referring to `Mode::Empty` here, rather than a results that happens to contain 0 items, and since we can never have a results in `Mode::Empty` here we can discount the possibility of encountering `none`?

Why does `Results::sum` return different results if it's in `Mode::Empty` vs containing 0 items? Aren't the two intended to provide the same behavior?